internal/pkg/cli/pipeline_init.go

```diff
@@ -237,30 +238,41 @@ func (opts *InitPipelineOpts) createBuildspec() (string, error) {
 	return path, nil
 }
 
-func (opts *InitPipelineOpts) selectEnvironments(addMore bool) error {
-	if addMore == false {
-		return nil
-	}
-
+func (opts *InitPipelineOpts) selectEnvironments() error {
+	var addEnv bool
+	envs := opts.listAvailableEnvironments()
 	addEnv, err := opts.prompt.Confirm(
 		pipelineAddEnvPrompt,
 		"Adds an environment that corresponds to a deployment stage in your pipeline. Environments are added sequentially.",
 	)
-
 	if err != nil {
 		return fmt.Errorf("failed to confirm adding an environment: %w", err)
 	}
-
-	var selectMoreEnvs bool
 	if addEnv {
-		selectMore, err := opts.selectEnvironment()
+		err := opts.selectEnvironment()
 		if err != nil {
 			return err
 		}
-		selectMoreEnvs = selectMore
+	}
+	for len(envs) != 0 && addEnv {
```

Comment:
This looks so much better! Thanks @iamhopaul123, I'm writing the way I'd have done for your reference. Essentially, it moves all the "unhappy" cases to the top and leaves only the happy case at the end:
```go
for {
	promptMsg := pipelineAddEnvPrompt
	promptHelpMsg := pipelineAddEnvHelpPrompt
	if len(opts.Environments) > 0 {
		promptMsg = pipelineAddMoreEnvPrompt
		promptHelpMsg = pipelineAddMoreEnvHelpPrompt
	}
	
	shouldAdd, err := opts.prompt.Confirm(promptMsg, promptHelpMsg)
        if err != nil {
		return fmt.Errorf("confirm adding an environment: %w", err)
        }
	if !shouldAdd {
		break	
	}
	if len(opts.listAvailableEnvironments()) == 0 {
		break	
	}
	if err := opts.selectEnvironment(); err != nil {
		return err
	}
}
```

Reply:
Thanks Efe, that's really enlightening and helpful!